rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -70,16 +76,16 @@ understand. We can always add more abstraction later if we need it.
 enum ExecutionState {
     Pending,
     Partition,
-    Write,
+    MaterializeAndWrite,
```

Comment:
Should these be two separate states?

Reply:
because of partitioning, there may be simultaneous materialization and segment writing occurring
we could add a synchronization point between the two steps if that's what you mean?

Reply:
Yes that is what I am asking, materialization seems lightweight enough to make this discrete but I am not sure if there are any valid reasons other than it feels cleaner.

Reply:
understood
I prefer to leave it as-is for now
after the third PR in this stack materialization, segment writing, committing, and flushing can all (theoretically) be running concurrently for different partitions
so if there's a good reason to have a sync point here (debugging? not sure how helpful it would be though) we should probably also add a sync point for all other steps